fix: Adjust min/max for 32f_s32f_convert_8i kernel

jdemel opened the pull request:
This kernel used `CHAR_MIN` to determine the minimum value which may be
0. It is expected that this value is -127. Thus, we move to `INT8_MIN`.

Fix #390

jdemel committed:
fix: Adjust min/max for 32f_s32f_convert_8i kernel

This kernel used `CHAR_MIN` to determine the minimum value which may be
0. It is expected that this value is -127. Thus, we move to `INT8_MIN`.

Fix #390

drmpeg commented:
Tested on armv7 and works good. I also verified that CHAR_MIN = 0 and CHAR_MAX = 255 since char is unsigned on ARM.

For consistency (even though they'll never execute on ARM), should the other instances of CHAR_MIN and CHAR_MAX in the module be changed to INT8_MIN and INT8_MAX?

jdemel committed:
fix: Use INT8_* instead of CHAR_*

For consistency and to prevent bugs, we move from CHAR_* to INT8_*
everywhere in this kernel.

jdemel commented:
I'd agree that they need to be updated too. Thanks for pointing this out.

michaelld reviewed (approved):
Please change the comment on line 28 : "char" -> "int". After that, LGTM & let's merge this.

jdemel committed:
docs: Fix conversion inaccuracy

michaelld commented:
Relevant CI passes. Maintainer approval. Important fix. Merging.
